python/src/iceberg/types.py

```diff
@@ -41,6 +45,12 @@ def __new__(cls, *args, **kwargs):
         return cls._instance
 
 
+class Truncatable(ABC):
```

Comment:
This should be parameterized because the `truncate` method always returns the type passed in, right?

Reply:
Thanks for pointing it out and that's a great point. Updated the code to use TypeVar and Generic for that.